.github/workflows/packer-build-and-publish.yml

```diff
@@ -233,11 +317,126 @@ jobs:
           fi
 
       - name: Publish release
+        id: publish
         uses: softprops/action-gh-release@c062e08bd532815e2082a85e87e3ef29c3e6d191 # v2.0.8
         with:
-          files: artifacts/**/*.json
+          files: |
+            artifacts/**/*.json
+            artifacts/**/*.md
           body: |
-            Release ${{ github.ref_name }} from workflow [#${{ github.run_id}}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
-          draft: ${{ needs.check-if-main.outputs.is-main != 'true' || contains(github.ref_name, '-') }}
-          make_latest: ${{ !(github.base_ref != 'main' || contains(github.ref_name, '-')) && steps.is_latest.outputs.value }}
+            Release ${{ github.ref_name }} from workflow [#${{ github.run_id }}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
+          draft: ${{ needs.check-main.outputs.is-main != 'true' || contains(github.ref_name, '-') }}
+          make_latest: ${{ !(needs.check-main.outputs.is-main != 'true' || contains(github.ref_name, '-')) && steps.is_latest.outputs.value }}
           generate_release_notes: true
+
+  share-and-copy-amis:
+    needs:
+      - check-main
+      - publish
+    runs-on: ubuntu-latest
+
+    env:
+      DEV_ACCOUNT: "654654387067"
+      PROD_ACCOUNT: "654654387067"
+      AWS_REGION: us-east-2
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Download manifest assets
```

Comment:
What if we merged this with the `publish` job? That one downloads the assets too, so we could avoid having to do it twice. The whole fetch-rename-publish could be done in one GitHub script actually, thinking about it.

Reply:
The publish job runs every time to create a release (draft or not) the copy job only runs if this is a prod release in which case we would need to copy the images to the prod account.

Publish downloads the assets either from the workflow or from a draft release to upload them on a new release.

This downloads only the manifest assets from a release to know which AMIs need to be copied over.

Reply:
We could merge the copy and publish jobs but the copy depends on if it's a prod release or not to run, while publish runs every time. 

The steps:
1. Assume role on source account
2. Share AMIs with destination account
3. Assume role on destination account
4. Copy AMIs over

Would all have a conditional to run on the publish job depending if it's a prod release or not and would all get skipped otherwise.

Reply:
Well you've implemented asset handling two times so it's a tradeoff of doing that versus having some `if` conditions if I understand correctly?

Reply:
I think merging both jobs won't change the aspect of having asset handling implemented twice, since both jobs handles assets differently, assets are downloaded from different places due to their availability on the workflow.

The publish job only downloads assets when a release doesn't exist yet, the assets are downloaded from the workflow after the build job generated and attached them to the workflow, so they can be uploaded as assets on the newly created release.

The share-and-copy-amis job will run on a different workflow, for a different tag, where the build job won't run, so those assets won't be present on the workflow, we have to download them from the existing release.

Both downloading assets from the workflow or from the release would be present even if we merged those jobs since both cases needed to be handled.